src/lib_json/json_value.cpp

```diff
@@ -848,8 +848,8 @@ bool Value::asBool() const {
   case uintValue:
     return value_.uint_ ? true : false;
   case realValue:
-    // This is kind of strange. Not recommended.
-    return (value_.real_ != 0.0) ? true : false;
+    // According to JavaScript language zero or NaN is regarded as false
+      return std::fpclassify(value_.real_) != FP_ZERO && std::fpclassify(value_.real_) != FP_NAN;
```

Comment:
Nit: the standard library is unclear on the complexity of the `std::fpclassify` macro.

For readability as well as avoiding duplicate calls (unclear whether the second call would be compiled out), I would recommend placing the fpclassify in a single call on a separate line, e.g.

```
case realValue: {
    const auto value_classification = std::fpclassify(value_.real_);
    return (value_classification != FP_ZERO) && (value_classification != FP_NAN);
}
```